src/lib.rs

```diff
@@ -892,16 +892,34 @@ impl<'ctx> Iterator for LocationRangeUnitIter<'ctx> {
 }
 
 fn path_push(path: &mut String, p: &str) {
-    if p.starts_with('/') {
+    if has_unix_root(p) || has_windows_root(p) {
         *path = p.to_string();
     } else {
-        if !path.ends_with('/') {
-            path.push('/');
+        let dir_separator = match (has_unix_root(path.as_str()), has_windows_root(path.as_str())) {
+            (true, _) => '/', // Unix path
+            (false, true) => '\\', // Windows path
+            (false, false) => '/', // Unknown path, so let's do it the unix way
+        };
+
+        if !path.ends_with(dir_separator) {
```

Comment:
Maybe we should allow either separator here? I've seen mingw tools generate paths that include both separators.

Reply:
Hmm, that seems to just be rust's prefix map which looks wrong (e.g. `/rustc/4fb7144ed159f94491249e86d5bbd033b5d60550\src\libcore\str`). This is probably fine how it is.